### PR TITLE
Adds fire ants

### PR DESCRIPTION
**Pull request by ShizCalev**
:cl: ShizCalev
add: Space ants will now turn into space fire ants when burned with fire.
/:cl:

### CODE REVIEW

**Comment by the-orange-cow** [comment]
Everything changed when the fire ants attacked...

**Comment by Wallemations** [comment]
And here I thought ants couldn't get any more diabolical

I highly recommend figuring out a way to differentiate them visually though.

Also technically this makes it possible to multiply ants exponentially by setting a small pile on fire then scooping it back up, if I'm not mistaken

**Comment by CursedBirb** [comment]
Make them more red?

**Comment by ShizCalev** [comment]
Now in a hill made of ash.

**Comment by Wallemations** [comment]
> Now in a hill made of ash. 

Metal.

File: code/game/objects/effects/decals/cleanable/misc.dm [hunk 1]
```diff
@@ -316,6 +316,26 @@
 	. = ..()
 	. += emissive_appearance(icon, "[icon_state]_light", alpha = src.alpha)
 
+/obj/effect/decal/cleanable/ants/fire_act(exposed_temperature, exposed_volume)
+	new /obj/effect/decal/cleanable/ants/fire(loc)
+	qdel(src)
```
Comment (Wallemations) [inline code/game/objects/effects/decals/cleanable/misc.dm:321]: Why not make the fire ants take their reagent number from the original ant pile by running the reagent amount through this `new` proc into the fire ant's `Initialize()`?

File: code/game/objects/effects/decals/cleanable/misc.dm [hunk 2]
```diff
@@ -316,6 +316,28 @@
...
+
+/obj/effect/decal/cleanable/ants/fire/update_ant_damage()
+	AddComponent(/datum/component/caltrop, min_damage = 15, max_damage = 25, flags = (CALTROP_NOCRAWL | CALTROP_NOSTUN | CALTROP_BYPASS_SHOES), soundfile = bite_sound)
+	update_icon(UPDATE_OVERLAYS)
+
+/obj/effect/decal/cleanable/ants/fire/fire_act(exposed_temperature, exposed_volume)
+	return
+
+/obj/effect/decal/cleanable/ants/fire/burn()
+	return
```
Comment (JohnFulpWillard) [inline code/game/objects/effects/decals/cleanable/misc.dm:339]: instead of this, can we just set ``resistance_flags = FIRE_PROOF``?

**Comment by Wallemations** [comment]
You updated it but it's still gonna be overwritten by the parent's `Initialize()`. The only reagents the ant piles are capable of holding are ants, so putting a variable in the fire ants' `Initialize()` and running the reagent amount through the `new` proc should fix it.

I also just realized that I overlooked the existence of `mapload` when I first coded these. Instead of doing all the shit I said above, you could instead just add an `if(mapload)` to the parent's `Initialize(mapload)` ONLY for the reagent_amount part. That would be way easier to do, and it would make me able to remove a particularly stupid check I put in later

**Pull request merged**